core/src/main/scala/org/apache/spark/scheduler/Task.scala

```diff
@@ -193,7 +192,7 @@ private[spark] object Task {
     dataOut.flush()
     val taskBytes = serializer.serialize(task).array()
```

Comment:
Maybe I'm just catching up here and you already discussed this, but `serialize` already returns a `ByteBuffer` right? why make it an array (does this call to `array()` need fixing?) and then write it into another one? it can be cloned directly, if one is worried about the caller modifying the buffer.